server/lib/maps/map-parse-worker.js

```diff
@@ -46,84 +46,80 @@ function createLobbyInitData(chk) {
   }
 }
 
-function generateImage(map, bwDataPath) {
-  if (bwDataPath !== '') {
-    // Create 1024x1024 images, or, if the map is not a square, have the larger of the
-    // dimensions be 1024 pixels.
-    let width
-    let height
-    if (map.size[0] > map.size[1]) {
-      width = 1024
-      height = map.size[1] * Math.floor(1024 / map.size[0])
-    } else {
-      height = 1024
-      width = map.size[0] * Math.floor(1024 / map.size[1])
+// Creates an image with the provided width and calculates height so the aspect ratio is preserved.
+function generateImage(map, bwDataPath, width = 1024) {
+  if (!bwDataPath) {
+    return Promise.resolve()
+  }
+
+  const aspectRatio = map.size[0] / map.size[1]
+  const height = Math.floor(width / aspectRatio)
```

Comment:
My gut instinct says a `round` is probably more appropriate here (although it probably makes a small difference)

Reply:
I also wonder if it might be less error-inducing to do this via a multiplication + divide, rather than 2 divides of smaller numbers. i.e. rearrange the formulas:

```
aspectRatio = actualWidth / actualHeight;
newHeight = desiredWidth / aspectRatio;
=>
newHeight = desiredWidth / (actualWidth / actualHeight)
=>
newHeight = desiredWidth * actualHeight / actualWidth
```